src/ol/geom/circle.js

```diff
@@ -223,7 +227,9 @@ ol.geom.Circle.prototype.setFlatCoordinates =
 
 
 /**
- * Set the radius of the circle. The radius is in the units of the projection.
+ * Set the radius of the circle in projected units.
+ * For EPSG:3857 or EPSG:4326 circles, this method
+ * does not set the radius in meters.
```

Comment:
Its the change of scale with latitude on 3857 that is the problem. This is adjusted for by e.g. the scale line control. The existing measure example gives the same answers for both its methods zoomed in near the equator. Its the change of scale with latitude that make the two methods give different results not the difference between calculating on the sphere or locally projected plane. The same issue applies to e.g. setting the circle radius.